corehq/apps/accounting/tests/base_tests.py

```diff
@@ -11,3 +15,53 @@ class BaseAccountingTest(TestCase):
     def setUpClass(cls):
         super(BaseAccountingTest, cls).setUpClass()
         Role.get_cache().clear()
+
+
+class BaseInvoiceTestCase(BaseAccountingTest):
+
+    is_using_test_plans = False
+    min_subscription_length = 3
+    is_testing_web_user_feature = False
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
move to `setUp` instead of `setUpClass`, and likewise with `tearDownClass`

Reply:
(additionally, better to move `tearDown` functionality to cleanup classes, so that cleanup actually executes if an error occurs)

Reply:
After working on this for a couple hours, it's looking pretty messy to try and move the subscription setup entirely out of the class method, based on various subclasses assuming (reasonably) that there is always a default subscription.

Rather than explicitly assigning subscriptions in each of those subclasses, I may end up instead, for test cases that need a unique subscription (rather than the default), explicitly _replacing_ that default subscription instead of adding a new one (which I believe was @jingcheng16 's concern in `test_community_invoice`, that we were just generating a new subscription but leaving still leaving the old one intact)

Reply:
Actually, that wouldn't quite work either, because then further test cases would be missing the expected class-subscription. It's specifically the `TestBillingAutoPay` classes that make heavy use of the class-subscription, so maybe I can restructure those a bit to work with a subscription set per-test case.